packages/jest-circus/src/utils.ts

```diff
@@ -177,11 +177,22 @@ export const callAsyncCircusFn = (
     // soon as `done` called.
     if (fn.length) {
       let returnedValue: unknown = undefined;
+
       const done = (reason?: Error | string): void => {
         // We need to keep a stack here before the promise tick
         const errorAtDone = new Error();
         // Use `Promise.resolve` to allow the event loop to go a single tick in case `done` is called synchronously
-        Promise.resolve().then(() => {
+        Promise.resolve().then(async () => {
+          const {seenDone} = getState();
+
+          if (!seenDone) {
+            await setState({...getState(), seenDone: true});
+          } else {
+            asyncError.message =
+              'Expected done to be called once, but it was called multiple times.';
+            return reject(asyncError);
```

Comment:
```suggestion
            errorAtDone.message =
              'Expected done to be called once, but it was called multiple times.';
            return reject(asyncError);
```

that way the stack points to the actual `done` call

Reply:
I should `return reject(errorAtDone)` then, right?